### PR TITLE
[tflchef] Introduce ModelChef

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce struct ModelChef to provide model chef functionality as beginning of refactoring to provide huge model generation.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13627

File: compiler/tflchef/core/src/ModelChef.cpp [hunk 1]
```diff
@@ -219,6 +189,20 @@ struct CookParams
   std::vector<std::string> &custom_code_vec;
   std::string noname;
 };
+#endif
+
+struct ModelChef
```
Comment (seanshpark) [inline compiler/tflchef/core/src/ModelChef.cpp:194]: `ModelChef` is introduce to replace `CookParams` and later to provide huge Buffer management.

File: compiler/tflchef/core/src/ModelChef.cpp [hunk 2]
```diff
@@ -219,6 +189,20 @@ struct CookParams
...
+{
+  std::unique_ptr<flatbuffers::FlatBufferBuilder> flatbuffer_builder;
+
+  std::vector<flatbuffers::Offset<::tflite::SignatureDef>> signdef_vec;
+  std::vector<flatbuffers::Offset<::tflite::Buffer>> buffer_vec;
+  std::vector<flatbuffers::Offset<::tflite::OperatorCode>> code_vec;
+  std::vector<flatbuffers::Offset<::tflite::SubGraph>> subgraph_vec;
+  std::map<tflite::BuiltinOperator, int32_t> builtin_code_map;
+  std::vector<std::string> custom_code_vec;
```
Comment (shs-park) [inline compiler/tflchef/core/src/ModelChef.cpp:203]: Is there any specific reason to remove the reference indicator(`&`)?

```suggestion
  std::unique_ptr<flatbuffers::FlatBufferBuilder> &flatbuffer_builder;

  std::vector<flatbuffers::Offset<::tflite::SignatureDef>> &signdef_vec;
  std::vector<flatbuffers::Offset<::tflite::Buffer>> &buffer_vec;
  std::vector<flatbuffers::Offset<::tflite::OperatorCode>> &code_vec;
  std::vector<flatbuffers::Offset<::tflite::SubGraph>> &subgraph_vec;
  std::map<tflite::BuiltinOperator, int32_t> &builtin_code_map;
  std::vector<std::string> &custom_code_vec;
```

**Review by shs-park — COMMENTED** [review]
I left a question, PTAL
=)

File: compiler/tflchef/core/src/ModelChef.cpp (hunk 2 above)
Comment (seanshpark) [reply compiler/tflchef/core/src/ModelChef.cpp:203]: > Is there any specific reason to remove the reference indicator(&)?

`CookParams` was to pass multiple parameters but
`ModelChef` will be used to hold these parameters as members.

**Review by shs-park — APPROVED** [review]
LGTM!
=)

**Pull request merged**